shell/platform/darwin/macos/framework/Source/FlutterEngine_Internal.h

```diff
@@ -127,11 +127,11 @@ typedef NS_ENUM(NSInteger, FlutterAppExitResponse) {
 /**
  * Dissociate the given view controller from this engine.
  *
- * Practically, since FlutterEngine can only be attached with one controller,
- * the given controller must be the default view controller.
- *
  * If the view controller is not associated with this engine, this call throws an
  * assertion.
+ *
+ * Practically, since FlutterEngine can only be attached with one controller for
+ * now, the given controller must be the current view controller.
```

Comment:
I would remove the "for now" here. There isn't enough context here to understand when/how this would change and I don't think it is important for understanding the functionality of the method.